backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -790,10 +805,41 @@ public FutureExperimentRunDAO(
                                     executor);
 
                             // Get datasets
-                            final var futureDatasets = datasetHandler.getArtifactsMap(ids);
+                            final var futureDatasetsMap = datasetHandler.getArtifactsMap(ids);
+                            final var filterDatasetsMap =
+                                futureDatasetsMap.thenCompose(
+                                    artifactMapSubtypes -> {
+                                      List<InternalFuture<Map<String, List<Artifact>>>>
+                                          internalFutureList = new ArrayList<>();
+                                      for (ExperimentRun.Builder builder : builders) {
+                                        internalFutureList.add(
+                                            privilegedDatasetsHandler
+                                                .filterAndGetPrivilegedDatasetsOnly(
+                                                    artifactMapSubtypes.get(builder.getId()),
+                                                    true,
+                                                    this::checkEntityPermissionBasedOnResourceTypes)
+                                                .thenCompose(
+                                                    artifacts ->
+                                                        InternalFuture.completedInternalFuture(
+                                                            Collections.singletonMap(
+                                                                builder.getId(), artifacts)),
+                                                    executor));
+                                      }
+                                      return InternalFuture.sequence(internalFutureList, executor)
+                                          .thenCompose(
+                                              maps -> {
+                                                Map<String, List<Artifact>> finalDatasetMap =
+                                                    new HashMap<>();
+                                                maps.forEach(finalDatasetMap::putAll);
+                                                return InternalFuture.completedInternalFuture(
+                                                    finalDatasetMap);
+                                              },
+                                              executor);
+                                    },
+                                    executor);
```

Comment:
This has nothing to do with creation, so the title is misleading and the PR more complex than it should be. We don't need to remove from this PR, but we have to stop doing this.

Reply:
Sorry about this :( . i have take care of this in other PRs but might be forgot in this. Updated PR title